### PR TITLE
fix(wallet): remove duplication nft_id when fetching nft metadata

### PR DESCRIPTION
**Pull request by nuo-xu**
Resolves https://github.com/brave/brave-browser/issues/41839

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Open wallet 
2. make a send tx that sends NFT `A` (doesn't matter it succeeded or not)
3. make another send tx that sends NFT `A`
4. observe no crash happens in the tx confirmation screen or tx status screen 
5. go to Activity Tab
6. observe tx list loads and no crash happens 
7. Close Wallet and reopen it again
8. go to Activity tab
9. observe tx list loads and no crash happens

### CODE REVIEW

File: ios/brave-ios/Sources/BraveWallet/Crypto/Stores/TransactionsActivityStore.swift [hunk 1]
```diff
@@ -227,6 +227,8 @@ class TransactionsActivityStore: ObservableObject, WalletObserverStore {
             $0.caseInsensitiveCompare(token.contractAddress) == .orderedSame
           })
         }
+      // remove duplicate
+      nftsWithoutMetadata = Array(Set(nftsWithoutMetadata))
```
Comment (StephenHeaps) [inline ios/brave-ios/Sources/BraveWallet/Crypto/Stores/TransactionsActivityStore.swift:231]: Shouldn't this just occur inside the helper function so the fix can apply elsewhere?
`func fetchNFTMetadata(tokens: [BraveWallet.BlockchainToken], ipfsApi: IpfsAPI) async -> [String: BraveWallet.NftMetadata]`

Comment (nuo-xu) [reply ios/brave-ios/Sources/BraveWallet/Crypto/Stores/TransactionsActivityStore.swift:231]: i know you gonna comment on that once i pushed the change lol
yeah i will move it to `fetchNFTMetadata` haha

Comment (nuo-xu) [reply ios/brave-ios/Sources/BraveWallet/Crypto/Stores/TransactionsActivityStore.swift]: updated at [99f8260](https://github.com/brave/brave-core/pull/26174/commits/99f8260f44088596dca2d95fe2549858b224438b)

**Review by StephenHeaps — APPROVED** [review]

**Comment by srirambv** [comment]
Verification passed on iPhone 13 with iOS 18.1 running 1.73 (46)
- Verified steps from PR description
- Verified no crash when opening activity list for same NFT transactions
- Verified able to open NFT transactions from the list

https://github.com/user-attachments/assets/89e4ee49-0e28-4255-a4e6-7522f4f22ee1

**Pull request merged**